spyder/plugins/completion/providers/languageserver/provider.py

```diff
@@ -567,41 +560,24 @@ def file_opened_closed_or_updated(self, filename: str, language: str):
     @on_conf_change
     def update_configuration(self, config):
         self.config = config
-        if running_under_pytest():
-            if not os.environ.get('SPY_TEST_USE_INTROSPECTION'):
-                return
         self.update_lsp_configuration()
 
     @on_conf_change(section='outline_explorer',
                     option=['group_cells', 'show_comments'])
     def on_pyls_spyder_configuration_change(self, option, value):
-        if running_under_pytest():
-            if not os.environ.get('SPY_TEST_USE_INTROSPECTION'):
-                return
         self.update_lsp_configuration()
 
     @on_conf_change(section='completions', option='enable_code_snippets')
     def on_code_snippets_enabled_disabled(self, value):
-        if running_under_pytest():
-            if not os.environ.get('SPY_TEST_USE_INTROSPECTION'):
-                return
         self.update_lsp_configuration()
 
     @on_conf_change(section='pythonpath_manager', option='spyder_pythonpath')
     def on_pythonpath_option_update(self, value):
-        # This is only useful to run some self-contained tests
-        if (
-            running_under_pytest()
-            and os.environ.get('SPY_TEST_USE_INTROSPECTION')
-        ):
-            self.update_lsp_configuration(python_only=True)
+        self.update_lsp_configuration(python_only=True)
 
```

Comment:
Why did you remove the `running_under_pytest()` condition?

Reply:
Oh, I overlooked the comment 😅 Will readd the `running_under_pytest()` validation and the comment :+1: